### PR TITLE
[ETL-686] Use S3 as GX validations store

### PR DESCRIPTION
**Pull request by philerooski**
A lot of files touched here, but most of those changes involve shuffling parameters around. The primary changes are:

## `src/glue/jobs/run_great_expectations_on_parquet.py`

- A large part of the GX data context configuration was moved to `src/glue/resources/great_expectations.yml`. This file is downloaded and formatted by `configure_gx_config`. The configuration of the datasource (`get_batch_request`, which now uses fluent datasources rather than the older datasource API) and the expectation suite (`add_expectations_from_json`) is still done in Python.
- In order to have a cumulative collection of validation reports in our docs, I had to change two things: persisting our validation store between job runs, which meant using S3 as the store backed rather than the local file system, and using checkpoints to update the data docs, rather than the `context.build_data_docs` method, which builds the entire site from scratch each time and can apparently take quite a while if we are adding to the data docs each day, judging from some GX issue reports. Checkpoints were necessary, since they allow us to use the `UpdateDataDocsAction` to update our data docs. As far as I can tell, there is no method to do this using a data context method. We also get a `StoreValidationResultAction` with checkpoints, so that saves us a step.

## `tests/test_run_great_expectations_on_parquet.py`

Because of the big refactor above, the tests changed a lot, as well.

## `src/scripts/manage_artifacts/artifacts.py`

I added the namespaced location of our data docs to the `--remove` option of this script, so feature branch data docs ought to be cleaned up upon branch deletion.

### CODE REVIEW

**Review by rxu17 — COMMENTED** [review]
Nice work! So much to redo just because of the non-persistent data context before :'). Just had some comments/clarification questions

File: src/glue/jobs/run_great_expectations_on_parquet.py
[the remark this answers is not in the document]
Comment (thomasyu888) [reply src/glue/jobs/run_great_expectations_on_parquet.py:1]: ^ These are great questions, that said, RECOVER is sunsetting in June, so we can table this for later

File: src/glue/jobs/run_great_expectations_on_parquet.py [hunk 1]
```diff
@@ -263,145 +178,103 @@ def read_json(
...
+    logger.info("get_batch_request")
+    batch_request = get_batch_request(
+        gx_context=gx_context, spark_dataset=spark_df, data_type=args["data_type"]
     )
-    context.build_data_docs(
-        site_names=["s3_site"],
+    logger.info("add_or_update_checkpoint")
+    # The default checkpoint action list is:
+    # StoreValidationResultAction, StoreEvaluationParametersAction, UpdateDataDocsAction
+    checkpoint = gx_context.add_or_update_checkpoint(
```
[the remark this answers is not in the document]
Comment (philerooski) [reply src/glue/jobs/run_great_expectations_on_parquet.py:270]: Yes, the `checkpoint.yml` file is still written locally by GX. The exact filename is dependent on the data type: in the case of Fitbit daily data it's written to `gx/checkpoints/fitbitdailydata-checkpoint.yml`. Although that's an irrelevant side effect of this job (it disappears once the job completes). The important thing is that the checkpoint points at the batch request, so it's convenient that we instantiate both the batch request and checkpoint dynamically in the code.

File: src/glue/jobs/run_great_expectations_on_parquet.py [hunk 2]
```diff
@@ -212,29 +132,24 @@ def get_spark_df(
...
-        data_type (str): data type name
-        run_id (RunIdentifier): contains the run name and
-            run time metadata of this batch run
+        spark_dataset (pyspark.sql.dataframe.DataFrame): A Spark dataframe
+        data_type (str): The data type
 
     Returns:
-        RuntimeBatchRequest: contains metadata for the batch run request
-            to identify this great expectations run
+        BatchRequest: A batch request which can be used in conjunction
```
[the remark this answers is not in the document]
Comment (philerooski) [reply src/glue/jobs/run_great_expectations_on_parquet.py:147]: We used to get our batch request from a deprecated "datasource" object, which was provided to us as a `RuntimeBatchRequest`. Now we are using a "fluent datasource", which returns a different object (`gx.datasource.fluent.batch_request.BatchRequest` to be exact). It doesn't have anything to do with preserving data.

File: src/glue/jobs/run_great_expectations_on_parquet.py [hunk 3]
```diff
@@ -263,145 +178,103 @@ def read_json(
...
-
-    # Create a ValidationResultIdentifier using the run_id, expectation suite, and batch identifier
-    validation_result_identifier = ValidationResultIdentifier(
-        expectation_suite_identifier=expectation_suite_identifier,
-        batch_identifier=batch_identifier,
-        run_id=run_identifier,
-    )
-
-    context.validations_store.set(validation_result_identifier, validation_result)
+    for data_type in expectations_data:
```
[the remark this answers is not in the document]
Comment (philerooski) [reply src/glue/jobs/run_great_expectations_on_parquet.py:202]: No, it's simpler than that. There's practically no overhead in adding the additional expectation suites. This saves us some additional complexity by not needing to pass the data type and add conditional logic to determine which expectations to add to the suite. But really, you could implement this either way. Either you add a bit of complexity and shave off 0.1% from the job execution time or you could do it the way I did it here.

File: tests/test_run_great_expectations_on_parquet.py [hunk 4]
```diff
@@ -171,60 +188,27 @@ def test_get_spark_df_has_expected_calls():
...
-    assert batch_request.data_connector_name == "runtime_data_connector"
-    assert batch_request.data_asset_name == "user_data-parquet-data-asset"
-    assert (
-        batch_request.batch_identifiers["batch_identifier"]
-        == "user_data_test_run_2023_batch"
+    batch_request = run_gx_on_pq.get_batch_request(
+        gx_context=gx_context, spark_dataset=spark_dataset, data_type=data_type
     )
-    assert batch_request.runtime_parameters["batch_data"] == spark_dataset
+    assert isinstance(batch_request, gx.datasource.fluent.batch_request.BatchRequest)
```
[the remark this answers is not in the document]
Comment (philerooski) [reply tests/test_run_great_expectations_on_parquet.py:197]: I'll add a test, although it's not important to the correctness of the job.

File: src/glue/jobs/run_great_expectations_on_parquet.py
[the remark this answers is not in the document]
Comment (philerooski) [reply src/glue/jobs/run_great_expectations_on_parquet.py:1]: > Testing expectation.json

I think this is covered pretty well by the existing test for `add_expectations_from_json`. As long as the JSON/dict we use as part of that test matches the format of `glue/resources/data_values_expectations.json` we can be reasonably confident that the expectations from the JSON are being added correctly to the expectation suite.

> Testing the number of versions in report
This is harder. Probably this works best as an integration test, although you could test components of this with a unit test that, for example, checks that running a checkpoint more than once updates the validation store.

**Review by rxu17 — APPROVED** [review]
LGTM!
Just a failing test: https://github.com/Sage-Bionetworks/recover/actions/runs/11637550526/job/32410945073

**Pull request merged**